### PR TITLE
created config file to route between production and development

### PR DESCRIPTION
**Pull request by MarlenAw**
Reference #25

### CODE REVIEW

**Review by MynahMarie — APPROVED** [review]

**Review by hoslack — APPROVED** [review]

**Comment by MarlenAw** [comment]
@hoslack, @MynahMarie apparently I didn't push everything..Sorry for the confusion! 
please check the last commit

```
Routing in production mode in order to deploy the app to heroku
```
- **Note**: I ran the command to create build folder

@hoslack please create a file called dev.js in the config folder and add your MONGO_URL to it

Once you guys approve and all, I'll deploy it tomorrow to heroku so Amira can see what we've done so far.

**Review by hoslack — APPROVED** [review]

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const host = process.env.HOST || 'localhost';
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+mongoose.connect(process.env.MONGO_URL);
+
+mongoose.connection.on('error', function(err) {
+  console.log('Mongoose default connection error: ' + err);
+});
```
Comment (MynahMarie) [inline server/app.js]: @MarlenAw maybe just some ES6 refactoring?

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+if (process.env.NODE_ENV === 'production') {
+  // Express will serve up production assets
+  // like our main.js file, or main.css file!
+  app.use('/api', reportRoutes);
+  app.use(express.static('client/build'));
+
+  // Express will serve up the index.html file
+  // if it doesn't recognize the route
+  const path = require('path');
```
Comment (MynahMarie) [inline server/app.js:37]: @MarlenAw path is already required at the top, no need to put it again...

**Review by MynahMarie — CHANGES_REQUESTED** [review]

**Review by MynahMarie — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -4,8 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "nodemon app.js"
+    "start": "node server/app.js",
+    "server": "nodemon node server/app.js",
```
Comment (Karyum) [inline package.json]: you dont need the second `node` here :sweat_smile:

File: package.json [hunk 4]
```diff
@@ -4,8 +4,12 @@
...
+    "client": "npm run start --prefix client",
+    "dev": "concurrently \"npm run server\" \"npm run client\"",
```
Comment (Karyum) [inline package.json]: i don't get it what is this script doing ? cause `npm run client` is doing `npm run start` which also starts the server

File: package.json [hunk 5]
```diff
@@ -4,8 +4,12 @@
...
+    "heroku-postbuild":
+      "NPM_CONFIG_PRODUCTION=false npm install --prefix client && npm run build --prefix client"
```
Comment (Karyum) [inline package.json]: what does --prefix do?

File: server/app.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+mongoose.connection.on('connected', () => {
+  console.log('Mongoose default connection open');
+  app.listen(port, () => {
+    console.log(`App listening on http://${host}:${port}`);
+  });
+});
+
+process.on('unhandledRejection', up => {
```
Comment (Karyum) [inline server/app.js]: NO lol, unhandled rejection means you did add `.catch()` on a promise which you always should!!!
if the either is about a deprecated promise library then it's something else [this is how](https://github.com/Automattic/mongoose/issues/4951#issuecomment-278081233) you would fix that problem with

```js
mongoose.Promise = require('bluebird')
```

File: server/app.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  throw up;
+}); //in Node 8.2.1 if theres an error, the errors will not crash the program. Instead a warning is printed..so this code
+// is here in order for that warning to go away and showing the error like usual
+
+if (process.env.NODE_ENV === 'production') {
```
Comment (Karyum) [inline server/app.js]: and if it's not production what will happen ? you wouldn't be able to use the api in development mode

**Review by Karyum — CHANGES_REQUESTED** [review]
# Big ones

File: package.json (hunk 3 above)
Comment (MarlenAw) [reply package.json]: I kinda do!! the first one is for heroku, the second one is for development mode

File: package.json (hunk 4 above)
Comment (MarlenAw) [reply package.json]: Again I need it so I can run it on development mode! look at "dev" script!

File: package.json (hunk 5 above)
Comment (MarlenAw) [reply package.json]: It's an instruction for scripts to build on Heroku

File: server/app.js (hunk 6 above)
Comment (MarlenAw) [reply server/app.js]: Well I have the latest version of node installed on my laptop! it throws warnings if I have errors on my code! It doesn't tell me where the error is unless I add this code here!

File: server/app.js (hunk 7 above)
Comment (MarlenAw) [reply server/app.js]: Added an else statement for dev

**Comment by Karyum** [comment]
@MarlenAw tiny change in the package-lock.json, me personally when it comes to conflicts in the package-lock.json i delete it :smile:

**Comment by MarlenAw** [comment]
@Karyum On it 😄 also isn't it better to add it to .gitignore from the start?

**Comment by Karyum** [comment]
@MarlenAw no package-lock.json will keep the versions of your dependencies that youre using so when let's say a dependency was update it won't crash your websire.
you also made another merge conflict by fixing the conflict :sweat_smile:

**Comment by MarlenAw** [comment]
@Karyum solved it :sweat:

**Comment by Karyum** [comment]
I really don't see why this branch was a blocker

**Pull request merged**